The test `array_layout_offsets` does not work

The default behaviour of the shader tests is to make the test pass if the shader compilation fails, to support machines that don't have the required opengl version. When you change this behaviour to panicking on a machine that should be able to run the test, the test `array_layout_offsets` fails. The compilation error is as follows:

```rust
thread 'array_layout_offsets' panicked at 'Shader compilation failed!: 
Compilation error in one of the shaders: 0(3) : error C0000: syntax error, unexpected ';', expecting "::" at token ";"', tests\shaders.rs:825
```

Changing:

```glsl
uniform layout(std140);

struct Foo {
  vec3 pos;
  vec3 dir;
  float speed;
};

uniform MyBlock {
 Foo data[256];
};
```

Into:
```glsl
struct Foo {
  vec3 pos;
  vec3 dir;
  float speed;
};

layout(std140)
uniform MyBlock{
  Bullet bullet_data[256];
};
```

Fixes the compilation error but produces the following runtime error:
```rust
thread '<main>' panicked at 'called `Result::unwrap()` on an `Err` value: UniformBlockLayoutMismatch {
  name: "BulletData",
  err: MemberMismatch { 
    member: "bullet_data",
    err: MemberMismatch { 
      member: "<dynamic array content>",
      err: MemberMismatch {
        member: "dir",
        err: OffsetMismatch {
          expected: 16,
          obtained: 12
        }
      }
    }
  }
}', ../src/libcore\result.rs:731
```

--- a/src/program/reflection.rs
+++ b/src/program/reflection.rs
@@ -850,7 +850,6 @@ fn introspection_output_to_layout<I>(elements: I) -> BlockLayout
     // ↓ actual body of `introspection_output_to_layout` starts here ↓
     let mut layout = BlockLayout::Struct { members: Vec::new() };
     for (name, offset, ty, array_size, top_level_array_size) in elements {
-        println!("{:?} {:?} {:?}", offset, array_size, top_level_array_size);
         process(&mut layout, &name, offset, ty, array_size, top_level_array_size);
     }
     layout

--- a/tests/shaders.rs
+++ b/tests/shaders.rs
@@ -793,20 +793,20 @@ fn array_layout_offsets() {
     let program = glium::Program::from_source(&display,
         "
             #version 330
-            uniform layout(std140);
 
             struct Foo {
-                vec3 pos;
-                vec3 dir;
+                vec2 pos;
+                vec2 dir;
                 float speed;
             };
 
+            layout(std140)
             uniform MyBlock {
                 Foo data[256];
             };
 
             void main() {
-                gl_Position = vec4(data[0].pos, 1.0);
+                gl_Position = vec4(data[0].pos, 0.0, 1.0);
             }
         ",
         "
@@ -837,18 +837,18 @@ fn array_layout_offsets() {
                 content: Box::new(glium::program::BlockLayout::Struct {
                     members: vec![
                         ("pos".to_string(), glium::program::BlockLayout::BasicType {
-                            ty: glium::uniforms::UniformType::FloatVec3,
+                            ty: glium::uniforms::UniformType::FloatVec2,
                             offset_in_buffer: 0,
                         }),
 
                         ("dir".to_string(), glium::program::BlockLayout::BasicType {
-                            ty: glium::uniforms::UniformType::FloatVec3,
-                            offset_in_buffer: 16,
+                            ty: glium::uniforms::UniformType::FloatVec2,
+                            offset_in_buffer: 8,
                         }),
 
                         ("speed".to_string(), glium::program::BlockLayout::BasicType {
                             ty: glium::uniforms::UniformType::Float,
-                            offset_in_buffer: 28,
+                            offset_in_buffer: 16,
                         }),
                     ],
                 }),

